gson/src/main/java/com/google/gson/JsonParser.java

```diff
@@ -26,7 +26,47 @@
 
 /**
  * A parser to parse JSON into a parse tree of {@link JsonElement}s.
+ * 
+ * <p>Here's an example of parsing from a string:
+ * 
+ * <pre>
+ * String json = "{\"key\": \"value\"}";
+ * JsonElement jsonElement = JsonParser.parseString(json);
+ * JsonObject jsonObject = jsonElement.getAsJsonObject();
+ * </pre>
  *
+ * <p>It can also parse from a reader:
+ * 
+ * <pre>
+ * try (Reader reader = new FileReader("C:\\my_json.json")) {
```

Comment:
To be bit more operating system agnostic, and to avoid "...json.json", what about using just `my-data.json` or similar as name?
```suggestion
 * try (Reader reader = new FileReader("my-data.json")) {
```

Maybe it should also explicitly specify the charset; otherwise users might encounter corruption of their data:
```suggestion
 * try (Reader reader = new FileReader("my-data.json", StandardCharsets.UTF_8)) {
```